app/api/v1/mediator/ballot.py

```diff
@@ -1,221 +1,262 @@
-    seed_hash = read_json_object(request.seed_hash, ElementModQ)
-    nonce: Optional[ElementModQ] = (
-        read_json_object(request.nonce, ElementModQ) if request.nonce else None
-    )
 
-    encrypted_ballots = []
-    current_hash = seed_hash
+    manifest, context, election_id = _get_election_parameters(election_id, request)
 
+    ballots = [SubmittedBallot.from_json_object(ballot) for ballot in request.ballots]
     for ballot in ballots:
-        encrypted_ballot = encrypt_ballot(
-            ballot, description, context, current_hash, nonce
+        if ballot.state == BallotBoxState.UNKNOWN:
+            raise HTTPException(
+                status_code=status.HTTP_405_METHOD_NOT_ALLOWED,
+                detail=f"Submitted ballot {ballot.object_id} must have a cast or spoil state",
+            )
+        validation_request = ValidateBallotRequest(
+            ballot=ballot.to_json_object(), manifest=manifest, context=context
         )
-        if not encrypted_ballot:
-            raise HTTPException(status_code=500, detail="Ballot failed to encrypt")
-        encrypted_ballots.append(encrypted_ballot)
-        current_hash = get_optional(encrypted_ballot.crypto_hash)
-
-    response = EncryptBallotsResponse(
-        encrypted_ballots=[ballot.to_json_object() for ballot in encrypted_ballots],
-        next_seed_hash=write_json_object(current_hash),
-    )
-    return response
+        _validate_ballot(validation_request)
 
+    return _submit_ballots(election_id, ballots)
 
-def handle_ballot(request: AcceptBallotRequest, state: BallotBoxState) -> Any:
-    ballot = CiphertextBallot.from_json_object(request.ballot)
-    description = Manifest.from_json_object(request.description)
-    internal_description = InternalManifest(description)
-    context = CiphertextElectionContext.from_json_object(request.context)
 
-    accepted_ballot = accept_ballot(
-        ballot,
-        state,
-        internal_description,
-        context,
-        DataStore(),
+@router.post("/validate", tags=[BALLOTS])
+def validate_ballot(
+    request: ValidateBallotRequest = Body(...),
+) -> BaseResponse:
+    """
+    Validate a ballot for the given election data
+    """
+    _validate_ballot(request)
+    return BaseResponse(
+        status=ResponseStatus.SUCCESS, message="Ballot is valid for election"
     )
 
-    return accepted_ballot
-
 
-def index_shares_by_ballot(
-    shares: List[DecryptionShare],
-) -> Dict[BALLOT_ID, Dict[GUARDIAN_ID, DecryptionShare]]:
-    """
-    Construct a lookup by ballot ID containing the dictionary of shares needed
-    to decrypt that ballot.
-    """
-    shares_by_ballot: Dict[str, Dict[str, DecryptionShare]] = {}
-    for share in shares:
-        ballot_shares = shares_by_ballot.setdefault(share.object_id, {})
-        ballot_shares[share.guardian_id] = share
+def _get_election_parameters(
+    election_id: Optional[str], request: BaseBallotRequest
+) -> Tuple[Manifest, CiphertextElectionContext, str]:
```

Comment:
in a future PR this function will try to query for the Manifest and Context from the mongo repository if they are not provided in the request body